File: BASICS/primeno.cpp
#include<iostream>
using namespace std;
bool prime (int n){
    if (n<2) return false;
    for (int i=2; i*i<=n; i++){
        if(n%i==0){
            return false; 
        }

        return true;
    }
}
int main(){
int num ;
cout<<"enter a number";
cin>>num;
if(prime(num)){
    cout<<"is prime";

}
else {
    cout<<"is nor prime";

}
}

File: BASIC 40/primeno.java
import java.util.*;

public class primeno {
    public static boolean isPrime(int n){
        boolean isPrime = true;

        if(n == 2){
            return true;
        }

        for(int i=2; i<=n;i++){
        if(n%2==0){
            isPrime = false;
            break;
        }
        
    }
    return isPrime;
   

}
   public static void main(String[] args) {
    Scanner sc = new Scanner(System.in);
    System.out.println("enter a number: ");
    int n= sc.nextInt();
    boolean idx = isPrime(n);
    System.out.println("the prime no: "+idx);
       
   }
}

